pkg/prometheus/promcfg_test.go

```diff
@@ -8783,3 +8783,205 @@ func TestPodMonitorWithNonDefaultScrapeClassRelabelings(t *testing.T) {
 	require.NoError(t, err)
 	golden.Assert(t, string(cfg), "podMonitorObjectWithNonDefaultScrapeClassWithRelabelings.golden")
 }
+
+func TestServiceMonitorWithDefaultScrapeClassMetricRelabelings(t *testing.T) {
```

Comment:
Awesome tests!

We could improve code readability here if we reduce code duplication using [table tests](https://dev.to/boncheff/table-driven-unit-tests-in-go-407b)